### PR TITLE
Read JSON context to avoid hard-coding of JSON field names

### PR DESCRIPTION
**Pull request by cmaumet**
This update adds support to read in the context information to select the fields in the JSON-LD document. This ensures that if different prefixes are used between two documents, the viewer will still run as it is based on the URIs rather than the prefix labels.

This PR will need to be tested further once #26 is merged but is open now to remind me of finishing up this update.

### CODE REVIEW

**Comment by TomMaullin** [comment]
Hi @cmaumet ,

I have just run tests for this branch on my machine and the following tests have failed:

Failure Summary:

`     Name                                          Failed  Incomplete  Reason(s)
    =============================================================================
     testDataSets/test_ex_spm_HRF_informed_basis     X         X       Errored.
    -----------------------------------------------------------------------------
     testDataSets/test_ex_spm_conjunction            X         X       Errored.
    -----------------------------------------------------------------------------
     testDataSets/test_ex_spm_contrast_mask          X         X       Errored.
    -----------------------------------------------------------------------------
     testDataSets/test_ex_spm_default                X         X       Errored.
    -----------------------------------------------------------------------------
     testDataSets/test_ex_spm_full_example001        X         X       Errored.
    -----------------------------------------------------------------------------
     testDataSets/test_ex_spm_group_ols              X         X       Errored.
    -----------------------------------------------------------------------------
     testDataSets/test_ex_spm_group_wls              X         X       Errored.
    -----------------------------------------------------------------------------
     testDataSets/test_ex_spm_partial_conjunction    X         X       Errored.
    -----------------------------------------------------------------------------
     testDataSets/test_ex_spm_temporal_derivative    X         X       Errored.
    -----------------------------------------------------------------------------
     testDataSets/test_ex_spm_thr_clustfwep05        X         X       Errored.
    -----------------------------------------------------------------------------
     testDataSets/test_ex_spm_thr_clustunck10        X         X       Errored.
    -----------------------------------------------------------------------------
     testDataSets/test_ex_spm_thr_voxelfdrp05        X         X       Errored.
    -----------------------------------------------------------------------------
     testDataSets/test_ex_spm_thr_voxelfwep05        X         X       Errored.
    -----------------------------------------------------------------------------
     testDataSets/test_ex_spm_thr_voxelunct4         X         X       Errored.
    -----------------------------------------------------------------------------
     testFeatures/checkViewerRunsSPM                 X         X       Errored.
    -----------------------------------------------------------------------------
     testFeatures/checkForTitle                      X         X       Errored.
    -----------------------------------------------------------------------------
     testFeatures/checkViewerRunsSPMwoMIP            X         X       Errored.`

The error message I get for each is the following:

` ---------
    Error ID:
    ---------
    'MATLAB:Containers:Map:NoKey'

    --------------
    Error Details:
    --------------
    Error using containers.Map/subsref
    The specified key is not present in this container.
    
    Error in changeNIDMtoTabDat (line 482)
                        tableTemp{n, 9} = str2double(get_value(peaksTemp{j}.(context('prov_value'))));
    
    Error in nidm_results_display (line 74)
                                       changeNIDMtoTabDat(graph,typemap,context,ids));
    
    Error in testFeatures/checkViewerRunsSPM (line 31)
                nidm_results_display(data_path);
================================================================================`

**Comment by cmaumet** [comment]
(Rebased on master)

**Comment by cmaumet** [comment]
```

result = 

  1×32 TestResult array with properties:

    Name
    Passed
    Failed
    Incomplete
    Duration
    Details

Totals:
   32 Passed, 0 Failed, 0 Incomplete.
   94.6325 seconds testing time.
```

**Comment by cmaumet** [comment]
This PR is now ready for review. Included updates:
 - Indexing of JSON fields based on URIs (but still via human-readable variables) 
 - Creation of an `isParametric` variable that will be useful later on to display results from non-parametric studies.

Non-regression tests:
```
result = 

  1×32 TestResult array with properties:

    Name
    Passed
    Failed
    Incomplete
    Duration
    Details

Totals:
   32 Passed, 0 Failed, 0 Incomplete.
   112.5771 seconds testing time.
```


@TomMaullin: your turn to review :)

File: lib/load_json_context.m [hunk 1]
```diff
@@ -0,0 +1,354 @@
...
+% viewer code (human-labels-from-code) with human labels found in the JSON
+% context (human-label-from-JSON)
+% This will ensure that if different labels are used in different documents
+% the viewer will still be able to read them (based on the URIs - unique).
+%
+% Authors: Thomas Maullin, Camille Maumet.
+%==========================================================================
+function context = load_json_context(json)
+    hkey_dict = human_key_dict();
+    not_found = hkey_dict.values;
```
Comment (TomMaullin) [inline lib/load_json_context.m]: Could you add a comment explaining why this variable was needed?

File: lib/load_json_context.m [hunk 2]
```diff
@@ -0,0 +1,354 @@
...
+    
+    hnms_dict = human_namespace_dict();
+
+    context = containers.Map();
+    
+    if isstruct(json)
+        json = {json};
+    end
+    
+    for i = 1:numel(json)
```
Comment (TomMaullin) [inline lib/load_json_context.m]: Please could you add a comment briefly explaining what is happening in this loop? (I.e. the subgraphs and then reading in the structure)

File: lib/changeNIDMtoTabDat.m [hunk 3]
```diff
@@ -267,28 +272,36 @@
...
-    if(ischar(volumeResels))
-        volumeResels = str2double(volumeResels);
+    if parametric
+        volumeResels = get_value(searchLinkedToCoord.(context('nidm_searchVolumeInResels')));
+        if(ischar(volumeResels))
+            volumeResels = str2double(volumeResels);
+        end
+    else
+        volumeResels = NaN;
     end
```
Comment (TomMaullin) [inline lib/changeNIDMtoTabDat.m]: Does SnPM not have any resel statistics?

Comment (cmaumet) [reply lib/changeNIDMtoTabDat.m]: Non-parametric FWE does not use random field theory which means that there is no need to compute resels. I've added a comment.

File: lib/load_json_context.m (hunk 1 above)
Comment (cmaumet) [reply lib/load_json_context.m]: This variable was unused. I have removed it. Thanks!

File: lib/load_json_context.m (hunk 2 above)
Comment (cmaumet) [reply lib/load_json_context.m]: Done.

**Comment by cmaumet** [comment]
I've made the requested changes. Non-regression tests:
```
result = 

  1×32 TestResult array with properties:

    Name
    Passed
    Failed
    Incomplete
    Duration
    Details

Totals:
   32 Passed, 0 Failed, 0 Incomplete.
   68.7582 seconds testing time.
```

**Comment by cmaumet** [comment]
I've made the requested changes. Non-regression tests:
```
result = 

  1×32 TestResult array with properties:

    Name
    Passed
    Failed
    Incomplete
    Duration
    Details

Totals:
   32 Passed, 0 Failed, 0 Incomplete.
   68.7582 seconds testing time.
```

**Pull request merged**